gpflow/misc.py

```diff
@@ -251,5 +251,55 @@ def _get_tensor_safe(name, index, graph):
     except KeyError:
         return None
 
+
+def swap_final_dims(X: tf.Tensor):
+    """
+    Equivalent to np.swapaxes(X, -2, -1).
+    Check https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.swapaxes.html for details.
+    """
+    K = tf.rank(X)
+    num_broadcasting_dims = tf.reduce_max([K - 2, 0])
+    perm = tf.concat([tf.range(num_broadcasting_dims), [K - 1, K - 2]], 0)
+    return tf.transpose(X, perm)
+
+
+def slice_final_dim(X: tf.Tensor, sl: slice, dim: int):
```

Comment:
Previous question disappeared: what dim is for, because it is not used in function.